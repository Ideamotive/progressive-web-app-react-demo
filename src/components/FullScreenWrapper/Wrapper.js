import React from 'react';
import styled from 'styled-components';

export default styled.div`
    position: fixed;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    z-index: 2;
  `;