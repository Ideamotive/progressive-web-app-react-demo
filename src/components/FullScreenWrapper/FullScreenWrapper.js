import React from 'react';
import PropTypes from 'prop-types';

// Styled Components
import Wrapper from './Wrapper'


class FullScreenWrapper extends React.Component {
  componentDidMount() {
    this.forceUpdate();
  }

  render() {
    const width = (this.fullScreenWrapper && this.fullScreenWrapper.offsetWidth) || 400;
    const height = (this.fullScreenWrapper && this.fullScreenWrapper.offsetHeight) || 400;
    return (
      <Wrapper innerRef={node => this.fullScreenWrapper = node}>
        {this.props.render(width, height)}
        {this.props.children}
      </Wrapper>
    );
  }
};

FullScreenWrapper.propTypes = {
  render: PropTypes.func.isRequired,
};

export default FullScreenWrapper;


