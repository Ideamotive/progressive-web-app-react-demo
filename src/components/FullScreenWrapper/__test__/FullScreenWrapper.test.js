import React from 'react';
import * as renderer from 'react-test-renderer';

import FullScreenWrapper from '../FullScreenWrapper';

describe('<FullScreenWrapper/> component', () => {
  test('should match to snapshot', () => {
    const tree = renderer
      .create(
        <FullScreenWrapper
          render={() => <div/>}
        >
          {'test'}
        </FullScreenWrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

