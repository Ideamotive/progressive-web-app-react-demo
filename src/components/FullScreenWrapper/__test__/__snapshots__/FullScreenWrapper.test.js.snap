// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullScreenWrapper/> component should match to snapshot 1`] = `
<div
  className="sc-bdVaJa dxncCL"
>
  <div />
  test
</div>
`;
